CDCgov/prime-reportstream
# OO/rhood/4318 Add Saliva to sender-automation.valueset

This PR contains modifications outlined in Ticket #4318 which covers the addition of Saliva as a specimen_type and the removal of cardinality: ONE from the specimen_source_site_code.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

**Test Steps:**
1. Process the attached CSV Test file using the command "curl -X POST -H "client:cadph-csvuploadertest" -H "content-type:text/csv" --data-binary "@./result_files/StandardCSV-Test-AL.csv"  "http://localhost:7071/api/reports"
3. Verify that the two messages with Saliva and Saliva specimen are code in the SPM.4 as "|119342007^Saliva specimen^SCT^^^^2.67|" and the SPM.8 is blank.

[StandardCSV-Test-AL.csv](https://github.com/CDCgov/prime-reportstream/files/8051724/StandardCSV-Test-AL.csv)


## Changes
- [x] Added Saliva and Sputum to the sender-automation.valueset under specimen_type
- [x] Removed cardinality: ONE from specimen_source_site_code


## Checklist
N/A

### Testing


### Process

## Fixes
*List GitHub issues this PR fixes*
- Closes #4219 


## To Be Done
*Create GitHub issues to track the work remaining, if any*


## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
